core/liblwm2m.h

```diff
@@ -332,10 +324,44 @@ typedef enum
     BINDING_UQS  // UDP queue mode plus SMS
 } lwm2m_binding_t;
 
+typedef enum    // data type of TLV_RESSOURCE for cond. observe
+{
+    LWM2M_DATATYPE_UNKNOWN = 0,
+    LWM2M_DATATYPE_STRING,
+    LWM2M_DATATYPE_INTEGER,
+    LWM2M_DATATYPE_FLOAT,
+    LWM2M_DATATYPE_BOOLEAN,
+    LWM2M_DATATYPE_OPAQUE,
+    LWM2M_DATATYPE_TIME
+} lwm2m_data_type_t;
+
+/** holds the attributes for object, instance and resources */
+typedef struct _attribute_data_ {
+  struct _attribute_data_ * next;    // next list element
+  uint16_t shortID;
+  time_t lastTransmission;  /**< holds the time stamp of the last transmission of this object, instance or resource */
+  time_t nextTransmission;  /**< holds the time for the next scheduled transmission */
+  lwm2m_uri_t uri;          /**< holds the URI of the object/instance/resource the attributes are valid for */
+  char* minPeriod;       /**< minimal update period in seconds, the object/instance/resource must not be sent more frequent */
+  char* maxPeriod;       /**< maximum update period, the object/instance/resource must be sent at least after this period, even if no change happend */
+  char* greaterThan;     /**< the resource has to be sent if this threshold is reached */
+  char* lessThan;        /**< the resource has to be sent if this threshold is underrun */
+  char* step;            /**< the resource has to be sent if it is changed by the given step */
+  char* oldValue;        /**< holds the last sent value of the resource if the resource is a numerical resource */
+  lwm2m_data_type_t resDataType;
```

Comment:
Ok, the benefit of is this solution is: The NULL pointer init shows unset mode. The question is valid, we also discussed that topic. The alternative would be a union construction with an additional set flag. May be better but a little bit more sophisticated.